rir/src/compiler/pir/closure.cpp

```diff
@@ -27,8 +27,7 @@ Promise* Closure::createProm() {
 
 Closure::~Closure() {
     for (auto p : promises)
-        if (p)
-            delete p;
+        delete p;
```

Comment:
I think I added this on purpose. If a promise get's unused, then the cleanup phase can delete it. This just sets it to null, otherwise all the other promises need to be renumbered (since we address them by index into this array).

Reply:
Ok but just peepholing this piece of code, the two versions are equivalent, no?
Because `delete nullptr;` is valid and is a no-op.

Reply:
oh, is it. didn't know, alright, then I'm ok